server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserService.kt

```diff
@@ -71,11 +69,50 @@ class PublicUserService(@Autowired private val publicUserRepository: PublicUserR
                     wins = it.wins,
                     losses = it.losses,
                     ties = it.ties
-                )
+                ),
             )
         }
     }
 
+    fun updateLeaderboard(leaderboardList: List<LeaderboardEntryDto>) {
+        val leaderboardSize = leaderboardList.size
+        for (i in 0 until leaderboardSize) {
+            val user = publicUserRepository.findByUsername(leaderboardList[i].user.username).get()
+            if (i < 0.2 * leaderboardSize) {
+                publicUserRepository.save(user.copy(tier = TierTypeDto.TIER1))
+            } else {
+                publicUserRepository.save(user.copy(tier = TierTypeDto.TIER2))
+            }
+        }
+    }
+
+    fun updateTempLeaderboardByTier() {
+        updateLeaderboard(getLeaderboardList())
+    }
+
+    fun updateLeaderboard() {
+        val leaderboardList = getLeaderboardList().sortedBy { it.stats.rating }.reversed()
```

Comment:
- Instead of sorting it manullay here, you can make use of the functions provided by `MongoRepositories` to query it.